app/javascript/mastodon/containers/status_container.js

```diff
@@ -111,6 +113,14 @@ const mapDispatchToProps = (dispatch, { intl }) => ({
     }
   },
 
+  onHide (status, history) {
+    dispatch(openModal('CONFIRM', {
+      message: <FormattedMessage id='confirmations.hide_status.message' defaultMessage='Are you sure to hide this status ?' />,
```

Comment:
It might make sense for the text to explain that a hidden status cannot be brought back.